backend/src/main/java/saci/domain/service/RoleService.java

```diff
@@ -12,8 +14,10 @@ public class RoleService {
 
     private RoleRepository roleRepository;
 
+
     public Role createRole(Role role) {
-        return roleRepository.save(role);
+        Optional<Role> optionalRole = roleRepository.findByName(role.getName());
+        return optionalRole.isPresent() ? null : roleRepository.save(role);
```

Comment:
Instead of return null here, take a look how you could use this exception

https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/client/HttpClientErrorException.Conflict.html

If you don't know how to throw exceptions in Java, check this
https://rollbar.com/guides/java/how-to-throw-exceptions-in-java/